### DISCUSSION TITLE
Any ideas about OpenType Math support?

### DISCUSSION THREAD

**Discussion by ainsley26** [rsms/inter#547] [Q&A] [answered]
Hello! 

I haven't ever done anything in the font space, so I have no understanding of the effort required, so here goes: _What's involved with adding support for the OpenType Math features?_

There are some fonts, such as [Fira](https://github.com/firamath/firamath)/[Lato](https://www.bing.com/search?q=lato+math&cvid=5d4fdccdaae845e1896b17076d568478&aqs=edge..69i57j0l2j46j0l2j46j0l2.1680j0j9&FORM=ANAB01&PC=U531)/[Noto](https://fonts.google.com/noto/specimen/Noto+Sans+Math) Math that do a pretty good job standalone, but seem a bit jarring since the style feels significantly different when used alongside Inter. Overall, there don't even seem to be that many fonts available for this task at all!

Just curious about it all and if there's something I could learn and help with, that might be fun too :)

Cheers,
A.

**Comment by rsms** [comment, marked as answer]
For glyphs with a direct mapping to Unicode codepoints it's simply a matter of drawing glyphs for the 12 different designs of Inter. It's labor intensive but not complicated.

Example from https://firamath.github.io/specimen.html:

[<img width="1007" alt="Screen Shot 2024-03-26 at 13 55 56" src="https://github.com/rsms/inter/assets/47046/3fd6d1bc-1a6a-4db6-a6d7-cae5e828a0d0">](https://firamath.github.io/specimen.html)

For advanced stuff like context-sensitive positioning and "ligatures", like something like this
<img width="75" alt="Screen Shot 2024-03-26 at 13 58 36" src="https://github.com/rsms/inter/assets/47046/4ddb1c6f-5c1a-49e6-b548-d11ab7b64dcd">
it gets much trickier as you'd need to write OpenType code _and_ design potentially hundreds of variations of existing glyphs for each permutation. The cost-benefit ratio for doing something like this is way off (high cost, low benefit, IMHO.)
